src/main/java/com/scalar/cassy/transferer/AzureFileUploader.java

```diff
@@ -0,0 +1,87 @@
+package com.scalar.cassy.transferer;
+
+import com.azure.storage.blob.BlobAsyncClient;
+import com.azure.storage.blob.BlobContainerAsyncClient;
+import com.azure.storage.blob.models.BlobProperties;
+import com.azure.storage.blob.models.BlobStorageException;
+import com.azure.storage.blob.models.ParallelTransferOptions;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Inject;
+import com.scalar.cassy.config.BackupConfig;
+import com.scalar.cassy.exception.FileIOException;
+import com.scalar.cassy.exception.FileTransferException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Mono;
+
+public class AzureFileUploader implements FileUploader {
+  private static final Logger logger = LoggerFactory.getLogger(AzureFileUploader.class);
+  private final BlobContainerAsyncClient blobContainerClient;
+
+  @Inject
+  public AzureFileUploader(BlobContainerAsyncClient blobContainerClient) {
+    this.blobContainerClient = blobContainerClient;
+  }
+
+  @Override
+  public void upload(List<Path> files, BackupConfig config) {
+    AtomicInteger count = new AtomicInteger();
+    count.getAndIncrement();
+    List<CompletableFuture> toBeUploaded = new ArrayList<>();
+    files.forEach(
+        p -> {
+          String key = BackupPath.create(config, p.toString());
+            String filePath = Paths.get(config.getDataDir(), p.toString()).toFile().getPath();
```

Comment:
Since the `requiresUpload` method takes a `Path` as parameter, it can be simplified.